src/matrix_projections.jl

```diff
@@ -396,12 +398,41 @@ function image_basis!(A::AbstractSparseMatrix)
     return A, p
 end
 
-function image_basis!(A::AbstractMatrix{T}) where {T<:Union{AbstractFloat,Complex}}
+function _orth!(M::AbstractMatrix{T}) where T<:Union{AbstractFloat, Complex}
+    F = svd!(convert(Matrix{T}, M))
+    M_rank = let m = T <: Complex ? 2eps(real(T)) : eps(T)
+        sum(F.S .> maximum(size(M)) * m)
+    end
+    return F.Vt, M_rank
+end
+
+function _orth!(M::AbstractSparseMatrix{T}) where T<:Union{AbstractFloat, Complex}
+    F = qr(M)
+    M_rank= rank(F)
+    result = let tmp = F.Q * Matrix(I, size(F.Q, 2), M_rank)
+        sparse(tmp[invperm(F.prow), :]')
```

Comment:
Also, as discussed would be nice to make the output type of invariant_vectors same as Upis since it can be regarded as Upis for the trivial subrep, that is SparseMatricSCS. This also includes making the type T in WedderburnDecomposition propagate to the results of invariant_vectors so that SparseMatricSCS{T} is returned (both for Upis and for invariant_vectors). @kalmarek

Reply:
@thinh-le yeah, I'm convinced that You're suggestion is a good solution, but let change the format of invariant vectors in a separate pull

Reply:
@kalmarek Thanks. I also believe that users would rarely use invariant vectors individually (if really wanted, views provide nice solution) but as matrix-vector multiplication (which is fast for both sparse and dense BLAS).

Also, I had a look at this issue https://github.com/kalmarek/SymbolicWedderburn.jl/issues/39 and for me it does not matter much if the user receive a row or column based Upis:
- adjoint is lazy
- matrix multiplication of Upi plays nice with adjoint (and fast when everything is sparse)

Maybe we should pick one that is fast for internals..